tests/events/tests_events_register_api.py

```diff
@@ -0,0 +1,97 @@
+        """
+        # create an unverifeid authenticated user
+        self.auth_unverified_user_email = "crash.test1.dummy@gmail.com"
+        self.auth_unverified_user_password = "test1.modelx"
+        self.auth_unverified_user = CustomUser.objects.create_user(
+            email = self.auth_unverified_user_email,
+            username = self.auth_unverified_user_email,
+            password = self.auth_unverified_user_password,
+            verified=False
+        )
+        
+        # create an verified authenticated user
+        self.auth_verified_user_email = "crash.test2.dummy@gmail.com"
+        self.auth_verified_user_password = "test2.modelx"
+        self.auth_verified_user = CustomUser.objects.create_user(
+            email = self.auth_verified_user_email,
+            username = self.auth_verified_user_email,
+            password = self.auth_verified_user_password,
+            verified=True
+        )
+        
+        
+        self.event = Event.objects.create(
+            id=1,
+            name= "test1",
+            venue = "test_venue",
+            time ="test_time",
+            details= "test_details",
+            details_html= "test.html",
+            cover_pic = "test_cover.jpeg",
+            icon= "test_icon.png",
+            email= "test@email.com",
+            flag = True,
+            year = 2019,
+        )
+
+    def test_fail_invalid_endpoint(self):
+        """
+        Test with invalid endpoint
+        """ 
+        post_register_event_api = "/events/register/twentieth/"
+        verified_auth_client = self.create_verified_auth_client()
+        response = verified_auth_client.post(post_register_event_api)
+        self.assertEqual(response.status_code,status.HTTP_404_NOT_FOUND)
+
+    def test_fail_unauthenticated_user(self):
+        """
+        Test with unauthenticated user
+        """
+        post_register_event_api = "/events/register/1/"
+        get_client = APIClient()
+        response = get_client.post(post_register_event_api)
+        self.assertEqual(response.status_code,status.HTTP_401_UNAUTHORIZED)
+
+    def test_fail_unverified_user(self):
+        """
+        Test with unverified user
+        """  
+        post_register_event_api = "/events/register/1/"
+        get_client = self.create_auth_client()
+        response = get_client.post(post_register_event_api)
+        self.assertEqual(response.status_code,status.HTTP_401_UNAUTHORIZED)
+
+
+    def test_pass_registeration(self):    
+        """
+        Test with registeration
+        """
+        post_register_event_api = "/events/register/1/"
+        verified_auth_client = self.create_verified_auth_client()
+        response = verified_auth_client.post(post_register_event_api)
+        self.assertEqual(response.status_code,status.HTTP_201_CREATED)
+
+    def test_fail_event_not_found(self):    
+        """
+        Test with event not found
+        """
+        post_register_event_api = "/events/register/0/"
```

Comment:
```suggestion
        post_register_event_api = "/events/register/12345/"
```
Lets use and extremely large number, so that something small like adding another event doesn't break the tests.